### PR TITLE
Separate commands for html generation of page+units and implementation

### PR DESCRIPTION
**Pull request by panglesd**
This was suggested in https://github.com/ocaml/odoc/pull/1185#discussion_r1701861980

The benefit is cleaner man pages, safer CLI validation.

This PR also takes this opportunity to do some cleanup, by removing the old and never used way of rendering sources:
  - Remove the "source dir" identifier. We have enough identifier types!
  - Remove the source_tree odoc unit.
  - Remove the `--source-root` option which is not anymore usable in odoc 3.

Once this is merged, I'll rebase #1185 on top and apply @Julow's comment!

### CODE REVIEW

**Review by Julow — APPROVED** [review]
Awesome! All this code that we didn't need is gone :)

**Comment by jonludlam** [comment]
Ooh I do like a good cleanup :-)

**Comment by jonludlam** [comment]
Just want to check though: should it be `odoc html-generate-source` or `src` rather than `odoc html-generate-impl`? Asking for opinions, I don't have a strong opinion.

**Comment by jonludlam** [comment]
Also we should have some negative tests - what happens when you do `odoc html-generate` on an impl file and vice-versa.

**Comment by panglesd** [comment]
In the odoc dev meeting, we decided to rename it to `compile-source` and use the source file as positional argument while the implementation file is passed as an option. It'll make it easier when including dune or other file in source rendering.

File: src/odoc/rendering.ml [hunk 1]
```diff
@@ -184,3 +145,38 @@ let targets_odoc ~resolver ~warnings_options ~syntax ~renderer ~output:root_dir
...
+          Renderer.traverse pages ~f:(fun filename _content ->
+              let filename =
+                Fpath.normalize @@ Fs.File.append root_dir filename
+              in
+              Format.printf "%a\n" Fpath.pp filename))
+        docs;
+      Ok ()
+  | Page_content _ | Unit_content _ | Asset_content _ ->
+      Error (`Msg "Expected an implementation unit")
+(* let docs = *)
```
Comment (Julow) [inline src/odoc/rendering.ml:168]: Brand new commented-out code :)

**Review by Julow — APPROVED** [review]
I think this is ready to be merged :) I've left some comments but I think it's fine to discuss them after this is merged.

File: src/odoc/bin/main.ml [hunk 2]
```diff
@@ -1038,6 +945,34 @@ end = struct
   end
 
   let targets ~docs = Targets.(cmd, info ~docs)
+
+  module Targets_source = struct
```
[the remark this answers is not in the document]
Comment (Julow) [reply src/odoc/bin/main.ml:949]: In fact we continued to maintain this command in Odoc 2 and there's a high chance that Odig uses it. Let's keep `-targets-sources`.

File: src/odoc/rendering.ml (hunk 1 above)
Comment (panglesd) [reply src/odoc/rendering.ml]: RIP

**Pull request merged**